libraries/AP_Scripting/lua_boxed_numerics.cpp

```diff
@@ -20,14 +20,14 @@ uint32_t coerce_to_uint32_t(lua_State *L, int arg) {
 
         int success;
         const lua_Integer v = lua_tointegerx(L, arg, &success);
-        if (success && v >= 0) {
+        if (success && v >= INT32_MIN) {
```

Comment:
```suggestion
        if (success) {
```

I'm pretty sure you are accidentally forcing a number of things into the float path at the moment? I was suggesting as long as we were passed an integer we just cast it, what you have here is ignoring many integer cases....